lib/restrict_admin_by_ip_middleware.rb

```diff
@@ -0,0 +1,32 @@
+class RestrictAdminByIpMiddleware
+  def initialize(app, allowed_ips)
+    @app = app
+    @allowed_ips = allowed_ips.map { |ip| IPAddr.new(ip) }
+  end
+
+  def call(env)
+    status, headers, body = @app.call(env)
+
+    if restricted?(env)
+      [403, headers, "Forbidden"]
```

Comment:
Rack expects the body to be enumerable so `"Forbidden"` will need to be wrapped in an array:
```suggestion
      [403, headers, ["Forbidden"]]
```

It's really strange that the spec passes though...